service/surf/apps/materials/utils.py

```diff
@@ -64,21 +64,30 @@ def add_extra_parameters_to_materials(user, materials):
     :param materials: array of materials
     :return: updated array of materials
     """
-    # perhaps this could be optimized by querying the database once for a list of materials
-    # however, we still want to add the 0 values for items we don't have in the database yet so this is easy for now
+
+    material_objects = list(Material.objects.filter(
+        external_id__in=(m["external_id"] for m in materials)
+    ))
+
+    educational_level_filters = MpttFilterItem.objects.filter(
+        name__in=(level for material in materials for level in material["educationallevels"])
+    ).distinct().select_related("title_translations").get_cached_trees()
+
     for m in materials:
+        material_object = next((x for x in material_objects if x.external_id == m["external_id"]), None)
```

Comment:
Dit kan dan wijzigen in:
`material_object = material_objects[m["external_id"]]`